gptel-curl.el

```diff
@@ -46,15 +46,9 @@ PROMPTS is the data to send, TOKEN is a unique identifier."
          (data (encode-coding-string
                 (json-encode (gptel--request-data prompts))
                 'utf-8))
-         (api-key
-          (cond
-           ((stringp gptel-api-key) gptel-api-key)
-           ((functionp gptel-api-key) (funcall gptel-api-key))
-           (t (setq gptel-api-key (read-passwd "OpenAI API key: ")))))
          (headers
           `(("Content-Type" . "application/json")
-            ("Authorization" . ,(concat "Bearer " api-key)))))
-
+            ("Authorization" . ,(concat "Bearer " (gptel--api-key))))))
     (push (format "-X%s" "POST") args)
```

Comment:
I am a bit unhappy with the code duplication of the two backends (url and curl). I think it would be okay if you restrict gptel to a single backend and require curl if that works much better for you.

Reply:
You're right, it's a bit of a mess right now. Let's just require curl for now -- but after we ensure that it works for everyone. Right now curl doesn't work for some users (#3), `url-retrieve` doesn't work for some (#2), and both backends don't work for others (#16).